### PR TITLE
Improved the writing documentation page.

### PR DESCRIPTION
**Pull request by pauloxnet**
#### Trac ticket number

"N/A"

#### Branch description

Improved the writing documentation page.

#### Checklist
- [x] This PR targets the `main` branch.
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [ ] I have checked the "Has patch" ticket flag in the Trac system.
- [ ] I have added or updated relevant tests.
- [ ] I have added or updated relevant docs, including release notes if applicable.
- [ ] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

File: docs/internals/contributing/writing-documentation.txt [hunk 1]
```diff
@@ -324,8 +340,9 @@ documentation:
     Five
     ^^^^
 
-* Use :rst:role:`:rfc:<rfc>` to reference RFC and try to link to the relevant
-  section if possible. For example, use ``:rfc:`2324#section-2.3.2``` or
+* Use :rst:role:`:rfc:<rfc>` to reference Request for Comments (RFC) and
```
Comment (thibaudcolas) [inline docs/internals/contributing/writing-documentation.txt:343]: yay! Great that we explain the acronyms. Could read a bit better?

```suggestion
* Use :rst:role:`:rfc:<rfc>` to reference a Request for Comments (RFC) and
```

File: docs/internals/contributing/writing-documentation.txt [hunk 2]
```diff
@@ -179,10 +184,21 @@ Link check
...
+links in the documentation are working. From the ``docs`` directory, run:
+
+.. console::
+
+     $ make linkcheck
+
+Output is printed to the terminal, but can also be found in
 ``_build/linkcheck/output.txt`` and ``_build/linkcheck/output.json``.
 
+.. admonition:: Be careful with the time
```
Comment (thibaudcolas) [inline docs/internals/contributing/writing-documentation.txt:196]: I like having that note but the heading isn’t super clear to me.

**Review by thibaudcolas — COMMENTED** [review]
Changes looking good, a few small things to suggest. I think you should explain the rationale for the change in how commands are displayed? Seems a bit arbitrary (unless this is part of a docs styleguide I’m not aware of)

File: docs/internals/contributing/writing-documentation.txt (hunk 2 above)
Comment (pauloxnet) [reply docs/internals/contributing/writing-documentation.txt:196]: Do you have suggestions for a clearer title?

**Comment by pauloxnet** [comment]
> ... I think you should explain the rationale for the change in how commands are displayed?

TL;DR I simply applied the presentation format of the 'make html' command to the other commands presented on the page.

Trying to read the guide to contributing to Django documentation from scratch as if I were a new contributor, I noticed that the command to generate the html documentation was very obvious, clear and compatible even for Windows users. The other commands presented on the page, however, were immersed in the text, not easy to locate and with the only variant for Unix users.

File: docs/internals/contributing/writing-documentation.txt (hunk 2 above)
Comment (sarahboyce) [reply docs/internals/contributing/writing-documentation.txt:196]: This can be a warning without a title

Comment (pauloxnet) [reply docs/internals/contributing/writing-documentation.txt]: Make sense to me. I removed the title, and updated the commit.

Comment (pauloxnet) [reply docs/internals/contributing/writing-documentation.txt]: > This can be a warning without a title

@sarahboyce I just find out that the `admonition` directive require a title :)

```
...writing-documentation.txt:196: ERROR: Error in "admonition" directive:
1 argument(s) required, 0 supplied.
```

So I added a tittle again in the code because the use of the `note` directive is discouraged in the same documents.

> To improve readability, use ``.. admonition:: Descriptive title`` rather tha  ``.. note::``

See https://docs.djangoproject.com/en/stable/internals/contributing/writing-documentation/#guidelines-for-restructuredtext-files

> I like having that note but the heading isn’t super clear to me.

@thibaudcolas I tried to add a clearer title

**Review by cliff688 — COMMENTED** [review]
Could you also link to `<coding-style-pre-commit>`) in line 309.

**Comment by pauloxnet** [comment]
> Could you also link to `<coding-style-pre-commit>` in line 309.

Thanks, I did it.

File: docs/internals/contributing/writing-documentation.txt (hunk 2 above)
Comment (sarahboyce) [reply docs/internals/contributing/writing-documentation.txt]: I think warnings can have no title

Comment (pauloxnet) [reply docs/internals/contributing/writing-documentation.txt]: > I think warnings can have no title (`.. warning::`)

Using `.. warning::` seems a good idea, thanks, I checked that it's widely used in the code. I just updated the PR. Thanks.

**Review by sarahboyce — APPROVED** [review]
Thank you @pauloxnet :star: these are nice updates
